tests/layers/l2_l4_inference_test.py

```diff
@@ -22,10 +22,20 @@
 """Tests for l2_l4_inference module."""
 
 import unittest
+import numpy
 
 from htmresearch.frameworks.layers import l2_l4_inference
 
 
+def _randomSDR(numOfBits, size):
+  """
+  Creates a random SDR for the given cell count and SDR size
+  :param numOfBits: Total number of bits
+  :param size: Number of active bits desired in the SDR
+  :return: list with active bits indexes in SDR
+  """
+  return list(numpy.random.permutation(numOfBits)[0:size])
```

Comment:
Have you considered `random.sample(xrange(numOfBits), size)`? I expect it to be dramatically more efficient. (This doesn't address the test failure)